x/distribution/keeper/delegation.go

```diff
@@ -190,5 +190,20 @@ func (k Keeper) withdrawDelegationRewards(ctx sdk.Context, val stakingtypes.Vali
 	// remove delegator starting info
 	k.DeleteDelegatorStartingInfo(ctx, del.GetValidatorAddr(), del.GetDelegatorAddr())
 
-	return coins, nil
+	if finalRewards.IsZero() {
```

Comment:
No state breaking changes. We just move event emission from the public method to the private one.

Reply:
would this effect gas?

Reply:
No. There are zero state R/W calls.